AEPCore/Sources/eventhub/EventHub.swift

```diff
@@ -268,6 +268,13 @@ final class EventHub {
         Log.debug(label: LOG_TAG, "Shared state created for \(EventHubConstants.NAME) with version \(version) and data: \n\(data as AnyObject)")
     }
 
+    /// Determines if there exists a registered extension with `extensionName`
+    /// - Parameter extensionName: The name of the extension to be looked up
+    /// - Returns: Returns true if this extension is currently registered with the `EventHub`, otherwise false
+    func isExtensionRegistered(extensionName: String) -> Bool {
+        return registeredExtensions.first(where: {$0.value.sharedStateName == extensionName}) != nil
+    }
```

Comment:
I would prefer not to expose this API yet as we don't encourage inter-extension dependencies. Another way to check that the edge extension is registered is by checking core's shared state.

Reply:
We already have use-cases where this API can be useful for example in Identity checking if Audience is present, and I believe there will be some in Analytics as well. I'd prefer not to have each extension duplicate logic for checking if an extension is registered (which will end up happening if we don't have an API provided) as parsing through the dictionary can be error-prone IMO.

Reply:
If we don't expose this API at this point, this allows us to re-evaluate the Edge dependency check later on for Identity and other extensions if needed. Analytics Edge doesn't use this approach as it is a new extension.

Reply:
I chatted with @shalehaha over the break and he's ok with exposing this API as well. It's not clear to me why we would need to re-evaluate the dependency check later on.